projects/plugins/boost/app/assets/src/js/react-components/pages/purchase-success/index.tsx

```diff
@@ -0,0 +1,98 @@
+import { getRedirectUrl } from '@automattic/jetpack-components';
+import { Button } from '@wordpress/components';
+import { createInterpolateElement, useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { requestImageAnalysis } from '../../../modules/image-size-analysis/store/isa-summary';
+import enableCloudCss from '../../../utils/enable-cloud-css';
+import { navigate } from '../../../utils/navigate';
+import Logo from '../../svg/jetpack-green';
+
+type PurchaseSuccessProps = {
+	isImageGuideActive: boolean;
+};
+
+const PurchaseSuccess: React.FC< PurchaseSuccessProps > = ( { isImageGuideActive } ) => {
+	useEffect( () => {
+		const mount = async () => {
+			enableCloudCss();
+
+			// If image guide is enabled, request a new ISA report.
+			if ( isImageGuideActive ) {
+				// Check if images can be resized.
+				if ( false !== Jetpack_Boost.site.canResizeImages ) {
+					await requestImageAnalysis();
+				}
+			}
+		};
+
+		mount();
+	}, [ isImageGuideActive ] );
```

Comment:
What's the advantage of wrapping it in a separate function compared to something like following:
```suggestion
	useEffect( () => {
		enableCloudCss();

		// If image guide is enabled, request a new ISA report.
		if ( isImageGuideActive && false !== Jetpack_Boost.site.canResizeImages ) {
			requestImageAnalysis();
		}
	}, [ isImageGuideActive ] );
```